src/it/scala/CirceDebeziumSpec.scala

```diff
@@ -0,0 +1,222 @@
+    field1s.traverse { field1 => for {
+      field2 <- Arbitrary.arbitrary[Short]
+      field3 <- Arbitrary.arbitrary[Long]
+      field4 <- Arbitrary.arbitrary[Boolean]
+      field5 <- Gen.identifier
+      field6 <- Arbitrary.arbitrary[Float]
+      field7 <- Arbitrary.arbitrary[Double]
+      field8 <- Arbitrary.arbitrary[Option[Int]]
+    } yield Debezium(field1, field2, field3, field4, field5, field6, field7, field8) }
+  }
+
+  def insertMany(dl: List[Debezium]): ConnectionIO[Int] = {
+    val sql = "INSERT INTO debezium (field1, field2, field3, field4, field5, field6, field7, field8) values (?, ?, ?, ?, ?, ?, ?, ?);"
+    Update[Debezium](sql).updateMany(dl)
+  }
+
+  def updateMany(dl: List[Debezium]): ConnectionIO[Int] = {
+    dl.traverse { d =>
+      Update[(Short, Long, Boolean, String, Float, Double, Option[Int], Int)]("UPDATE debezium SET field2 = ?, field3 = ?, field4 = ?, field5 = ?, field6 = ?, field7 = ?, field8 = ? WHERE field1 = ?;").run((d.field2, d.field3, d.field4, d.field5, d.field6, d.field7, d.field8, d.field1))
+    }.map(_.sum)
+  }
+
+  val genDoobieProg: Gen[IO[Int]] = for {
+    count   <- Gen.choose(1, 100)
+    field1s <- Gen.sequence[List[Int], Int]((1 to count).map(Gen.const))
+    ins     <- genDebeziums(field1s)
+    perm     = Random.shuffle(field1s)
+    indices <- Gen.delay(perm.take(Random.nextInt(count) + 1))
+    ups     <- genDebeziums(indices)
+    insert  <- Gen.delay(insertMany(ins))
+    update  <- Gen.delay(updateMany(ups))
+    delete  <- Gen.delay(sql"DELETE FROM debezium;".update.run)
+  } yield (insert.transact(transactor), update.transact(transactor), delete.transact(transactor)).mapN((a, b, c) => a + b + c)
+
+  val connectorName = "debezium-postgresql"
+  val dbServerName  = "debezium"
+
+  lazy val config = json"""{
+    "name": $connectorName,
+    "config": {
+      "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+      "database.hostname": ${postgres.networkAliases(0)},
+      "database.port": ${pgPort.toString},
+      "database.user": $pgUser,
+      "database.password": $pgPW,
+      "database.server.id": "184054",
+      "database.server.name": $dbServerName,
+      "database.dbname": $pgUser,
+      "table.whitelist": "public.debezium",
+      "tasks.max": "1",
+      "database.history.kafka.bootstrap.servers": $kafkaUrl,
+      "database.history.kafka.topic": "dbhistory.debezium",
+      "database.history.skip.unparseable.ddl": "true",
+      "include.schema.changes": "true"
+    }
+  }"""
+
+  lazy val consumerSettings = ConsumerSettings[IO, DebeziumKey[Int], Option[DebeziumValue[Debezium]]]
+    .withAllowAutoCreateTopics(false)
+    .withAutoOffsetReset(AutoOffsetReset.Earliest)
+    .withBootstrapServers(kafka.bootstrapServers)
+    .withGroupId("group")
+
+  "Interacting with Debezium" - {
+    "DebeziumPayload constructs correctly from real Debezium JSON" in {
+//      forAll (genDoobieProg) { doobie =>
```

Comment:
How come this doesn't work?

Reply:
It probably does. Do we want to try it? It will make the test take a long time if it does work.

Reply:
Is that just because it creates so many? We can change the default settings of that

Reply:
OK, I reinstated it, with a minor refactoring to create the connector outside the loop. I don't know that it takes too long. Do you want to try it?